frontend/src/hooks/Instance/useInstanceCreate.ts

```diff
@@ -0,0 +1,179 @@
+        dispatch?.({ type: INSTANCE_ACTIONS.START_FETCH })
+        try {
+            const response = await InstanceService.instanceInstanceDetails()
+            dispatch?.({
+                type: INSTANCE_ACTIONS.SET_INSTANCE_DETAILS,
+                payload: response.data
+            })
+            dispatch?.({ type: INSTANCE_ACTIONS.SET_ERROR, payload: null })
+        } catch (err) {
+            dispatch?.({
+                type: INSTANCE_ACTIONS.SET_ERROR,
+                payload: "Failed to load instance details. Please try again."
+            })
+            console.error("Error fetching instance details:", err)
+        } finally {
+            dispatch?.({ type: INSTANCE_ACTIONS.SET_LOADING, payload: false })
+        }
+    }, [dispatch])
+
+    // Update form handlers with useCallback
+    const handleOsTypeSelect = useCallback((osType: OsType) => {
+        setFormData(prev => ({ ...prev, os_type: osType }))
+    }, [])
+
+    const handleInstancePlanSelect = useCallback((instancePlan: InstancePlan) => {
+        setFormData(prev => ({ ...prev, instance_plan: instancePlan }))
+    }, [])
+
+    const handleRootPasswordChange = useCallback((password: string) => {
+        setFormData(prev => ({ ...prev, root_password: password }))
+    }, [])
+
+    const handleInstanceNameChange = useCallback((name: string) => {
+        setFormData(prev => ({ ...prev, instance_name: name }))
+    }, [])
+
+    // Get unique image names with useMemo
+    const uniqueImageNames = useMemo(() => {
+        if (!instanceDetails?.os_image) return []
+        return [...new Set(instanceDetails.os_image.map(img => img.os_image_name))]
+    }, [instanceDetails?.os_image])
+
+    // Get available versions for selected image with useMemo
+    const availableVersions = useMemo(() => {
+        if (!selectedImageName || !instanceDetails?.os_image) return []
+        
+        return instanceDetails.os_image
+            .filter(img => img.os_image_name === selectedImageName)
+            .map(img => img.os_image_version)
+    }, [selectedImageName, instanceDetails?.os_image])
+
+    // Handle image name selection
+    const handleImageNameSelect = useCallback((imageName: string) => {
+        setSelectedImageName(imageName)
+        setSelectedVersion("") // Reset version when image changes
+    }, [])
+
+    // Handle version selection
+    const handleVersionSelect = useCallback((version: string) => {
+        setSelectedVersion(version)
+        
+        // Find the complete OsType object
+        const selectedOs = instanceDetails?.os_image.find(
+            img => img.os_image_name === selectedImageName && img.os_image_version === version
+        )
+        
+        if (selectedOs) {
+            handleOsTypeSelect(selectedOs)
+        }
+    }, [instanceDetails?.os_image, selectedImageName, handleOsTypeSelect])
+
+    // Check if all required fields are filled
+    const isFormValid = useMemo(() => {
+        return !!(
```

Comment:
would recommend more specific check like A !== null/undefined though. boolean in javascript are inherently prone to type coercion